### DISCUSSION TITLE
futures-preview crate is pulled in even on stable toolchain

### DISCUSSION THREAD

**Discussion by hntd187** [sagebind/isahc#109] [General]
If I do a check on my crates this fails on stable because isahc is pulling in futures-preview even on stable. There is already a nightly feature in the cargo can those crates be gated behind it?

**Comment by sagebind** [comment]
That is correct, futures-preview is always used, but there's no reason why futures-preview won't work on stable. Disabling it via a feature would not be possible because futures-preview is an integral piece of Isahc's architecture. Most of Isahc's users are also on stable Rust and it works fine.

What problems are you having with futures-preview being pulled in exactly?

**Comment by hntd187** [comment]
When I do a compilation check on stable, it bombs saying futures-preview is not not allowed on stable yet

https://dev.azure.com/toshi-search/toshi-search/_build/results?buildId=407

**Comment by hntd187** [comment]
and I guess just to be clear, the crate works and builds if I ignore this, but I'm not even sure how this is possible in the first place since to my understanding a crate built on nightly like futures-preview shouldn't work at all on stable. Unless I'm misunderstanding something .

**Comment by sagebind** [comment]
Ah, that is because you need to use Rust 1.36.0 or newer, it looks like you are using 1.35.0. The `std::future` module (which the futures-preview crate relies on) was stabilized in 1.36.0.

Until Isahc reaches 1.0, the only supported Rust version I can guarantee is the latest stable release; though I try to avoid relying on brand-new features if I can. Though admittedly, when async/await stabilizes soon, an upcoming Isahc release will probably have a hard dependency on it.

**Comment by hntd187** [comment]
lemme bump the version required to 1.36, this is probably not a big deal since it's so close to stabilization anyways.

**Comment by sagebind** [comment]
1.36 is already stable, if that's what you were asking. The latest stable is 1.38. 1.39 is the next release that includes async/await. Though to be clear, async/await is _not_ currently a dependency of Isahc, though it _will_ become a dependency in the future (after it is stable at the very least).

**Comment by hntd187** [comment]
Oh I meant async/await on stable, sorry for being unclear. Yea, I just can't read here, there is nothing we need to do I think sorry for the confusion.
